backend/eventloop/shared_resource_loop/sharedresourceloop_repositorycredential_test.go

```diff
@@ -0,0 +1,283 @@
+						if reflect.DeepEqual(sanitizeCondition(resourceCondition), sanitizeCondition(existingCondition)) {
+							conditionExists = true
+							break
+						}
+					}
+					if !conditionExists {
+						fmt.Println("GitOpsDeploymentCondition:", conditionExists, "/ Expected:", expectedRepoCredStatus.Conditions, "/ Actual:", repoCred.Status.Conditions)
+						break
+					}
+				}
+				return conditionExists
+
+			}, BeTrue())
+		}
+
+		It("should update an existing condition if it has changed", func() {
+			gitopsDeploymentRepositoryCredentialCR.Spec.Secret = "test"
+			repoCredStatus := managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialStatus{
+				Conditions: []*metav1.Condition{
+					{
+						Type:   managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred,
+						Reason: managedgitopsv1alpha1.RepositoryCredentialReasonSecretNotSpecified,
+						Status: metav1.ConditionTrue,
+					}, {
+						Type:   managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryUrl,
+						Reason: managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+						Status: metav1.ConditionFalse,
+					}, {
+						Type:   managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryCredential,
+						Reason: managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+						Status: metav1.ConditionFalse,
+					},
+				},
+			}
+
+			gitopsDeploymentRepositoryCredentialCR.Status = repoCredStatus
+			Expect(k8sClient.Status().Update(ctx, gitopsDeploymentRepositoryCredentialCR)).To(BeNil())
+
+			expectedRepoCredStatus := managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialStatus{
+				Conditions: []*metav1.Condition{
+					{
+						Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred,
+						Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+						Status:  metav1.ConditionTrue,
+						Message: "repository not found",
+					}, {
+						Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryUrl,
+						Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+						Status:  metav1.ConditionFalse,
+						Message: "repository not found",
+					}, {
+						Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryCredential,
+						Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+						Status:  metav1.ConditionFalse,
+						Message: "repository not found",
+					},
+				},
+			}
+
+			err := UpdateGitopsDeploymentRepositoryCredentialStatus(ctx, gitopsDeploymentRepositoryCredentialCR, k8sClient, &corev1.Secret{}, log.FromContext(ctx))
+			Expect(err).To(BeNil())
+
+			Expect(gitopsDeploymentRepositoryCredentialCR).Should(SatisfyAll(haveErrOccurredConditionSet(expectedRepoCredStatus)))
+		})
+
+		It("shouldn't update an existing condition if it hasn't changed", func() {
```

Comment:
In order to tell if the condition has changed, we can check using lastTransitionTime, and verify it was not updated for any of the conditions.